File: PYTHON/Motherboard.py
class Motherboard:
    def __init__(self, socket_type="", chipset="", merk="", nama=""):
        self.socket_type = socket_type
        self.chipset = chipset
        self.merk = merk
        self.nama = nama

    def set_socket_type(self, socket_type):
        self.socket_type = socket_type

    def set_chipset(self, chipset):
        self.chipset = chipset

    def get_socket_type(self):
        return self.socket_type

    def get_chipset(self):
        return self.chipset

    def get_merk(self):
        return self.merk

    def get_nama(self):
        return self.nama

File: CPP/Motherboard.cpp
// Motherboard.cpp
#pragma once
#include <iostream>
#include <string>
#include "Komponen.cpp"

using namespace std;

class Motherboard : public Komponen
{
private:
    string socketType;  // Jenis soket CPU (misalnya: LGA1151, AM4)
    string chipset;     // Chipset yang digunakan oleh motherboard
public:
    Motherboard() {}

    Motherboard(string socketType, string chipset, string merk, string nama) : Komponen(merk, nama)
    {
        this->socketType = socketType;
        this->chipset = chipset;
    }

    void setSocketType(string socketType)
    {
        this->socketType = socketType;
    }

    void setChipset(string chipset)
    {
        this->chipset = chipset;
    }

    string getSocketType()
    {
        return this->socketType;
    }

    string getChipset()
    {
        return this->chipset;
    }

    ~Motherboard() {}
};
